Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Commands/SendDeletionProcessApprovalReminder/HandlerTests.cs

```diff
@@ -0,0 +1,197 @@
+        var mockPushNotificationSender = A.Fake<IPushNotificationSender>();
+
+        A.CallTo(() => mockIdentitiesRepository.FindAllWithDeletionProcessWaitingForApproval(A<CancellationToken>._, A<bool>._))
+            .Returns(new List<Identity> { identity });
+
+        var handler = CreateHandler(mockIdentitiesRepository, mockPushNotificationSender);
+
+        // Act
+        await handler.Handle(new SendDeletionProcessApprovalReminderCommand(), CancellationToken.None);
+
+        // Assert
+        A.CallTo(() => mockPushNotificationSender.SendNotification(A<IdentityAddress>.That.Matches(i => i == identity.Address), A<DeletionProcessWaitingForApprovalReminderPushNotification>._, A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+        A.CallTo(() => mockIdentitiesRepository.Update(A<Identity>.That.Matches(i =>
+                i.Address == identity.Address
+                && i.DeletionProcesses.FirstOrDefault(d => d.Status == DeletionProcessStatus.WaitingForApproval)!.ApprovalReminder3SentAt == utcNow
+            ), A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async Task Only_sends_second_reminder_when_first_reminder_wasnt_sent_on_the_same_run()
+    {
+        // Arrange
+        var identity = TestDataGenerator.CreateIdentityWithDeletionProcessWaitingForApproval(deletionProcessStartedAt: DateTime.Parse("2000-01-01"));
+
+        var utcNow = DateTime.Parse("2000-01-06");
+        SystemTime.Set(utcNow);
+
+        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        var mockPushNotificationSender = A.Fake<IPushNotificationSender>();
+
+        A.CallTo(() => mockIdentitiesRepository.FindAllWithDeletionProcessWaitingForApproval(A<CancellationToken>._, A<bool>._))
+            .Returns(new List<Identity> { identity });
+
+        var handler = CreateHandler(mockIdentitiesRepository, mockPushNotificationSender);
+
+        // Act
+        await handler.Handle(new SendDeletionProcessApprovalReminderCommand(), CancellationToken.None);
+
+        // Assert
+        A.CallTo(() => mockPushNotificationSender.SendNotification(A<IdentityAddress>.That.Matches(i => i == identity.Address), A<DeletionProcessWaitingForApprovalReminderPushNotification>._, A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+        A.CallTo(() => mockIdentitiesRepository.Update(A<Identity>.That.Matches(i =>
+                i.Address == identity.Address
+                && i.DeletionProcesses.FirstOrDefault(d => d.Status == DeletionProcessStatus.WaitingForApproval)!.ApprovalReminder1SentAt == null
+                && i.DeletionProcesses.FirstOrDefault(d => d.Status == DeletionProcessStatus.WaitingForApproval)!.ApprovalReminder2SentAt == utcNow
+            ), A<CancellationToken>._))
+            .MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async Task Only_sends_third_reminder_when_no_other_reminder_was_sent_on_the_same_run()
```

Comment:
I just realized that the test name isn't accurate - or is it? 

I think the correct name should be something like "Does_not_send_reminder_1_and_2_when_3_has_to_be_sent_as_well". What do you think?

Similarly for the test before and of course the grace period reminder tests as well.

Reply:
I agree, this is probably a leftover name from the previous logic we had implemented.
I'll update the respective test cases.